src/rgw/rgw_admin.cc

```diff
@@ -8141,9 +8141,11 @@ int main(int argc, const char **argv)
     if (max_entries < 0)
       max_entries = 1000;
 
+    const auto& gen = bucket_info.layout.logs.back().gen;
```

Comment:
these BILOG_ admin commands should all take a --gen option, and use `find_if(bucket_info.layout.logs...)` to get the right layout. only default to `bucket_info.layout.logs.back()` when that option isn't given

Reply:
when we give an option to pass --gen into cli, how should the user be aware of what gen to specify? We will have to provide the gen as an output in some command. From bucket stats command may be? But then again, we might only be providing the latest gen.

Reply:
it probably makes sense to add a new command like 'bucket layout' that dumps the bucket layout structure

Reply:
ack